Estuviste cerca @SergioRibera 

```rust
 error: this `impl` can be derived
  --> crates/core/src/lib.rs:14:1
   |
14 | / impl Default for TestResult {
15 | |     fn default() -> Self {
16 | |         Self { bad_addrs: 0 }
17 | |     }
18 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derivable_impls
   = note: `-D clippy::derivable-impls` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::derivable_impls)]`
   = help: remove the manual implementation...
help: ...and instead derive it
   |
10 + #[derive(Default)]
11 | pub struct TestResult {
   |

error: could not compile `memsos_core` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.
```

--- a/crates/core/src/lib.rs
+++ b/crates/core/src/lib.rs
@@ -7,14 +7,9 @@ use crate::test::pattern;
 use core::fmt::Arguments;
 use core::ops::{Add, AddAssign};
 
+#[derive(Default)]
 pub struct TestResult {
     pub bad_addrs: u64,
-}
-
-impl TestResult {
-    pub fn new() -> Self {
-        Self { bad_addrs: 0 }
-    }
 }
 
 impl Add for TestResult {

--- a/crates/core/src/test/pattern.rs
+++ b/crates/core/src/test/pattern.rs
@@ -3,7 +3,7 @@ use rand::{RngCore, SeedableRng};
 use rand_chacha::ChaCha20Rng;
 
 pub fn run_test_own_address<M: Mem>(mem: &M, region: &MemoryRegion) -> TestResult {
-    let offset_region = mem.parse(&region);
+    let offset_region = mem.parse(region);
     let mut bad_addrs = 0;
 
     for addr in offset_region.start..offset_region.end {
@@ -22,7 +22,7 @@ pub fn run_test_own_address<M: Mem>(mem: &M, region: &MemoryRegion) -> TestResul
 }
 
 pub fn run_test_rand_num<M: Mem>(mem: &M, region: &MemoryRegion) -> TestResult {
-    let offset_region = mem.parse(&region);
+    let offset_region = mem.parse(region);
     // TODO: make this dynamic
     // https://github.com/RustLangES/memsos/pull/8
     let mut rand = ChaCha20Rng::seed_from_u64(12381293);

--- a/kernel/src/main.rs
+++ b/kernel/src/main.rs
@@ -107,7 +107,7 @@ fn kernel_main(boot_info: &'static mut BootInfo) -> ! {
         &text!("Made with love by Rust Lang Es")
     );
 
-    let mut test_result = TestResult::new();
+    let mut test_result = TestResult::default();
 
     for region in regions.iter() {
         if region.kind != MemoryRegionKind::Usable {

--- a/kernel/src/main.rs
+++ b/kernel/src/main.rs
@@ -37,7 +37,9 @@ fn kernel_main(boot_info: &'static mut BootInfo) -> ! {
     let info = framebuffer.info();
     let buffer = framebuffer.into_buffer();
 
-    let Some(mem_offset) = physical else { loop {} };
+    let Some(mem_offset) = physical else {
+        panic!("no physical memory")
+    };
 
     let memory_writer = MemWriter::create(*mem_offset);
 
@@ -137,6 +139,7 @@ fn kernel_main(boot_info: &'static mut BootInfo) -> ! {
         )
     );
 
+    #[allow(clippy::empty_loop)]
     loop {}
 }
 

